### PR TITLE
feat: Add LinearArrayPool

### PR DESCRIPTION
**Pull request by ebariche**
## PR Type

What kind of change does this PR introduce?
- Perf

### <samp>🤖 Generated by Copilot at 7dd1acd</samp>

This pull request introduces a new `LinearArrayPool` class that is a custom array pool optimized for dependency property values. It also refactors the `DependencyPropertyDetails` and `DependencyPropertyDetailsCollection` classes to use the new array pool and improve the performance and memory usage of the dependency property system. Additionally, it renames the files and namespaces related to the array pool feature to use the `Buffers` name instead of the `Buffer` name.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UWP/Buffers/LinearArrayPool.cs [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+		private static IArrayPoolPlatformProvider _platformProvider = new DefaultArrayPoolPlatformProvider();
+
+		static LinearArrayPool()
+		{
+			_automaticMemoryManagement = WinRTFeatureConfiguration.ArrayPool.EnableAutomaticMemoryManagement;
+
+			_canUseMemoryManager = _automaticMemoryManagement && _platformProvider.CanUseMemoryManager;
+
+			_defaultAutomaticBucketSize = WinRTFeatureConfiguration.ArrayPool.DefaultAutomaticMaxNumberOfArraysPerBucket;
+			_defaultBucketSize = WinRTFeatureConfiguration.ArrayPool.DefaultMaxNumberOfArraysPerBucket;
```
Comment (Youssef1313) [inline src/Uno.UWP/Buffers/LinearArrayPool.cs:33]: Can we inline these and avoid the static constructor?

File: src/Uno.UWP/Buffers/LinearArrayPool.cs [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+			for (var x = 0; x < _buckets.Length; x++)
+			{
+				_buckets[x].Trim(memoryUsage, now);
+			}
+
+			return true;
+		}
+
+		private class Bucket
```
Comment (Youssef1313) [inline src/Uno.UWP/Buffers/LinearArrayPool.cs:130]: Consider sealing this class. It should result in a more efficient code. The difference might not be large compared to other parts of the code, it's a simple change anyway

File: src/Uno.UWP/Buffers/LinearArrayPool.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UWP/Buffers/LinearArrayPool.cs:33]: There will be a static constructor anyways, it probably won't change much.

File: src/Uno.UWP/Buffers/LinearArrayPool.cs [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Windows.System;
+
+namespace Uno.Buffers
+{
+	internal class LinearArrayPool<T>
```
Comment (jeromelaban) [inline src/Uno.UWP/Buffers/LinearArrayPool.cs:11]: Let's add tests like this one: https://github.com/unoplatform/uno/blob/78790e74a14caefa81175d7a8f5a8039f1724a4d/src/Uno.UI.Tests/Buffers/Given_ArrayPool.cs#L15

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**